base_rest/restapi.py

```diff
@@ -333,3 +333,77 @@ def to_json_schema(self, service, direction):
         if self._unique_items is not None:
             json_schema["uniqueItems"] = self._unique_items
         return json_schema
+
+
+class FormDataPart(object):
+    def __init__(self, name=None):
+        """
+        :param name: The name of the form-data field
+        """
+        self._name = name
+
+    def to_openapi_part_property(self, service, direction):
+        return {}
+
+
+class BinaryFormDataPart(FormDataPart):
+    def to_openapi_part_property(self, service, direction):
+        return {"type": "string", "format": "binary"}
+
+
+class StringFormDataPart(FormDataPart):
+    def to_openapi_part_property(self, service, direction):
+        return {"type": "string"}
+
+
+class JsonFormDataPart(FormDataPart):
+    def __init__(self, name, validator):
+        """This allows to create a part of a multipart/form-data endpoint.
+        :param validator:  an instance of RestMethodParam
+        """
+        super().__init__(name)
+        self._validator = validator
+        if not hasattr(validator, "to_json_schema"):
+            raise Exception("The validator should provide 'to_json_schema'")
+
+    def to_openapi_part_property(self, service, direction):
+        schema = self._validator.to_json_schema(service, direction)
+        return schema
+
+
+class MultipartFormData(RestMethodParam):
+    def __init__(self, parts):
+        """This allows to create multipart/form-data endpoints.
+        :param parts:  list of FormDataPart
+        """
+        if not isinstance(parts, list):
+            parts = [parts]
+        self._parts = parts
+
+    def to_openapi_properties(self, service, direction):
+        properties = {}
+        for part in self._parts:
+            properties[part._name] = part.to_openapi_part_property(service, direction)
+        return properties
+
+    def to_multipart_content_schema(self, service, direction):
+        return {
+            "multipart/form-data": {
+                "schema": {
+                    "type": "object",
+                    "properties": self.to_openapi_properties(service, direction),
+                }
+            }
+        }
+
+    def from_params(self, service, params):
+        return params
+
+    def to_openapi_requestbody(self, service):
+        return {"content": self.to_multipart_content_schema(service, "input")}
+
+    def to_openapi_responses(self, service):
+        return {"200": {"content": self.to_multipart_content_schema(service, "output")}}
+
+    def to_response(self, service, result):
+        return result
```

Comment:
What do you expect as result? How do you validate-it?

Reply:
I removed this possibility. i don't see a use case where we would want to return a multipart-formdata from the API so I think this would be better to not maintain it.